domainservice/src/test/kotlin/ua/marchenko/artauction/auction/mapper/AuctionMapperProtoTest.kt

```diff
@@ -0,0 +1,167 @@
+    ) {
+        // WHEN THEN
+        val exception = assertThrows<IllegalArgumentException> {
+            mongoBid.toBidProto()
+        }
+        Assertions.assertEquals(errorMessage, exception.message)
+    }
+
+    @Test
+    fun `should return BigDecimalProto from BigDecimal`() {
+        // GIVEN
+        val bigDecimal = BigDecimal.valueOf(Random.nextDouble(1.0, 100.0))
+        val expectedBigDecimal = BigDecimalProto.newBuilder().also {
+            it.scale = bigDecimal.scale()
+            it.intVal = bigDecimal.unscaledValue().toBigIntegerProto()
+        }.build()
+
+        // WHEN
+        val result = bigDecimal.toBigDecimalProto()
+
+        // THEN
+        assertEquals(expectedBigDecimal, result)
+    }
+
+    @Test
+    fun `should return BigIntegerProto from BigInteger`() {
+        // GIVEN
+        val bigInteger = BigInteger.valueOf(Random.nextLong(1L, 100L))
+        val expectedBigInteger = BigIntegerProto.newBuilder().also {
+            it.value = ByteString.copyFrom(bigInteger.toByteArray())
+        }.build()
+
+        // WHEN
+        val result = bigInteger.toBigIntegerProto()
+
+        // THEN
+        assertEquals(expectedBigInteger, result)
+    }
+
+    @Test
+    fun `should return MongoAuction from AuctionProto`() {
+        // GIVEN
+        val protoAuction = AuctionProtoFixture.randomAuctionProto()
+        val expectedAuction = MongoAuction(
+            id = protoAuction.id.toObjectId(),
+            artworkId = protoAuction.artworkId.toObjectId(),
+            startBid = protoAuction.startBid.toBigDecimal(),
+            startedAt = protoAuction.startedAt.toLocalDateTime(),
+            finishedAt = protoAuction.finishedAt.toLocalDateTime(),
+            buyers = protoAuction.buyersList.map { it.toBid() }
+        )
+
+        // WHEN
+        val result = protoAuction.toMongoAuction()
+
+        // THEN
+        assertEquals(expectedAuction, result)
+    }
+
+    @Test
+    fun `should return Bid from BidProto`() {
+        // GIVEN
+        val protoBid = AuctionProtoFixture.randomBidProto()
+        val expectedBid = MongoAuction.Bid(
+            buyerId = protoBid.buyerId.toObjectId(),
+            bid = protoBid.bid.toBigDecimal()
+        )
+
+        // WHEN
+        val result = protoBid.toBid()
+
+        // THEN
+        assertEquals(expectedBid, result)
+    }
+
+    companion object {
+        @JvmStatic
+        fun errorMongoBidToErrorMessage(): List<Arguments> = listOf(
+            Arguments.of(MongoAuction.Bid.random(buyerId = null), "Buyer id cannot be null"),
+            Arguments.of(MongoAuction.Bid.random(bid = null), "Bid cannot be null"),
+        )
+
+        @JvmStatic
+        fun errorMongoAuctionToErrorMessage(): List<Arguments> = listOf(
+            Arguments.of(MongoAuction.random(id = null), "Auction id cannot be null"),
+            Arguments.of(MongoAuction.random(artwork = MongoArtwork.random(id = null)), "Artwork id cannot be null"),
+        )
+    }
```

Comment:
better to use  @CsvSource

Reply:
not in this case, because it is a pair of MongoAuction object and error message